quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HttpPropertyMappers.java

```diff
@@ -130,68 +141,60 @@ public static PropertyMapper<?>[] getHttpPropertyMappers() {
-
-            config.keyStoreFile = Configuration.getOptionalKcValue(HttpOptions.HTTPS_KEY_STORE_FILE.getKey()).map(Paths::get);
-            config.keyStorePassword = Configuration.getOptionalKcValue(HttpOptions.HTTPS_KEY_STORE_PASSWORD.getKey());
-            config.keyStoreFileType = Configuration.getOptionalKcValue(HttpOptions.HTTPS_KEY_STORE_TYPE.getKey());
-            config.keyStoreProvider = Configuration.getOptionalValue("quarkus.http.ssl.certificate.key-store-provider");
-            config.keyStoreAlias = Configuration.getOptionalValue("quarkus.http.ssl.certificate.key-store-alias");
-            config.keyStoreAliasPassword = Configuration.getOptionalValue("quarkus.http.ssl.certificate.key-store-alias-password");
-            config.keyStoreAliasPasswordKey = Configuration.getOptionalValue("quarkus.http.ssl.certificate.key-store-alias-password-key");
-            config.keyStoreKeyAlias = Configuration.getOptionalValue("quarkus.http.ssl.certificate.key-store-key-alias");
-            config.keyFiles = Optional.empty();
-            config.files = Optional.empty();
-
-            try {
-                TlsUtils.computeKeyStoreOptions(config, config.keyStorePassword, config.keyStoreAliasPassword);
-            } catch (IOException e) {
-                throw new PropertyException("Failed to load 'https-key-store' material.", e);
-            } catch (IllegalArgumentException e) {
+        try {
+            TlsUtils.computeKeyStoreOptions(config, config.keyStorePassword, config.keyStoreAliasPassword);
+        } catch (IOException e) {
+            throw new PropertyException("Failed to load 'https-key-' material: " + e.getClass().getSimpleName() + " " + e.getMessage(), e);
+        } catch (IllegalArgumentException e) {
+            if (e.getMessage().contains(QUARKUS_HTTPS_KEY_STORE_FILE_TYPE)) {
                 throw new PropertyException("Unable to determine 'https-key-store-type' automatically. " +
                         "Adjust the file extension or specify the property.", e);
             }
-        }
-
-        if (!enabled) {
-            Optional<String> value = Configuration.getOptionalKcValue(HttpOptions.HTTPS_CERTIFICATE_FILE.getKey());
-
-            if (value.isEmpty()) {
-                value = Configuration.getOptionalValue("quarkus.http.ssl.certificate.key-store-file");
-            }
-
-            if (value.isEmpty()) {
-                throw new PropertyException(Messages.httpsConfigurationNotSet());
-            }
+            throw new PropertyException(e.getMessage(), e);
         }
     }
 
-    private static BiFunction<String, ConfigSourceInterceptorContext, String> validatePath(String key) {
-        return (value, context) -> Environment.isWindows() && value != null && value.equals(context.proceed(key).getValue()) ? value.replace("\\", "/") : value;
+    private static BiFunction<String, ConfigSourceInterceptorContext, String> transformPath(String key) {
```

Comment:
Providing the 'key' param seems confusing when it's not even used.

```suggestion
    private static BiFunction<String, ConfigSourceInterceptorContext, String> transformPath() {
```

Reply:
Simplified to just be a function, rather than returning a function. As far as I can tell the check of the context.forward value when the value isn't null isn't necessary - that would imply the usage of quarkus.properties, but that usage would be overriden by the value coming PropertyMappingInterceptor.